frontend/degree-plan/components/Search/SearchPanel.tsx

```diff
@@ -81,10 +81,10 @@ const SearchPanelHeader = styled(PanelHeader)`
 
 interface SearchPanelProp {
     activeDegreeplanId: DegreePlan["id"] | null;
-    setSearchedRuleId: (arg0: number) => void;
+    setSearchRuleId: (arg0: number | null) => void;
 }
 
-export const SearchPanel = ({ activeDegreeplanId, setSearchedRuleId }: SearchPanelProp) => {
+export const SearchPanel = ({ activeDegreeplanId, setSearchRuleId }: SearchPanelProp) => {
```

Comment:
why do we need to explicitly pass this down as a prop? could we just get it from the context?

Reply:
context is used on the next few lines

Reply:
Yea legacy code I will delete it